grafana-plugin/src/models/heartbeat/heartbeat.ts

```diff
@@ -70,4 +70,20 @@ export class HeartbeatStore extends BaseStore {
       };
     });
   }
+
+  @action.bound
+  async resetHeartbeat(id: Heartbeat['id']) {
+    const response = await makeRequest(`${this.path}${id}/reset`, { method: 'POST' });
+
+    if (!response) {
+      return;
+    }
+
+    runInAction(() => {
+      this.items = {
+        ...this.items,
+        [response.id]: response,
+      };
+    });
+  }
```

Comment:
I'd like to promote in our frontend codebase shifting the logic from view layer (components) into store layer (MobX), flatten components and leverage Aspect Oriented Programming. To contribute to this approach in this PR we can:

models/heartbeat.ts
```
  @WithGlobalNotification({ success: 'Heartbeat has been reset' })
  @action.bound
  async resetHeartbeatAndRefetchIntegration(
    heartbeatId: Heartbeat['id'],
    integrationId: ApiSchemas['AlertReceiveChannel']['id']
  ) {
    const response = await makeRequest(`${this.path}${heartbeatId}/reset`, { method: 'POST' });

    if (!response) {
      return;
    }

    await this.rootStore.alertReceiveChannelStore.fetchItemById(integrationId);

    runInAction(() => {
      this.items = {
        ...this.items,
        [response.id]: response,
      };
    });
  }
```

containers/IntegrationContainers/IntegrationHeartbeatForm/IntegrationHeartbeatForm.tsx
```
  async function onReset() {
    await heartbeatStore.resetHeartbeatAndRefetchIntegration(heartbeatId, alertReceveChannelId);
    onClose();
  }
```